airbyte-container-orchestrator/src/main/java/io/airbyte/container_orchestrator/AsyncStateManager.java

```diff
@@ -4,37 +4,98 @@
 
 package io.airbyte.container_orchestrator;
 
+import com.google.common.annotations.VisibleForTesting;
 import io.airbyte.workers.process.AsyncKubePodStatus;
 import io.airbyte.workers.process.KubePodInfo;
+import io.airbyte.workers.storage.DocumentStoreClient;
+import jakarta.inject.Singleton;
+import java.lang.invoke.MethodHandles;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * The state manager writes the "truth" for states of the async pod process. If the store isn't
  * updated by the underlying pod, it will appear as failed.
- *
+ * <p>
  * It doesn't have a single value for a state. Instead, in a location on cloud storage or disk, it
  * writes every state it's encountered.
  */
-public interface AsyncStateManager {
```

Comment:
Curious, why remove the interface? Because there is no implementation?

Reply:
I called this out in the description on this PR as well
> combine AsyncStateManager interface with DefaultAsyncStateManager class into AsyncStateManager class

There was only a single implementation of this interface and I didn't see any benefit to having this additional layer of abstraction.